File: Java/time-conversion.java
class Result {
    public static String timeConversion(String s) {
        String[] arr = s.split("(?=[A-Z]{2}$)");
        String[] time = arr[0].split(":");
        int hr = Integer.parseInt(time[0]);
        String ampm = arr[1];
        if("PM".equals(ampm)){
            if(hr!=12){
                hr+= 12;
            }
        }
        else{
            if(hr==12){
                hr = 0;
            }
        }
        time[0] = String.format("%02d",hr);
        return String.join(":", time);
    }
}

File: Python/time-conversion.py
def timeConversion(s):
    time = s[:-2].split(":")
    hr = int(time[0])
    ampm = s[-2:]
    
    if ampm == 'PM':
        if hr != 12:
            hr+=12
    else:
        if hr == 12:
            hr = 0
    
    time[0] = f"{hr:02d}"
    return str.join(":",time)